rpc/server_auth.go

```diff
@@ -36,11 +37,19 @@ const (
 // as authentication metadata.
 type JWTClaims struct {
 	jwt.RegisteredClaims
-	CredentialsType CredentialsType   `json:"rpc_creds_type,omitempty"`
-	AuthMetadata    map[string]string `json:"rpc_auth_md,omitempty"`
+	AuthCredentialsType CredentialsType   `json:"rpc_creds_type,omitempty"`
+	AuthMetadata        map[string]string `json:"rpc_auth_md,omitempty"`
 }
 
-// Entity entity from the claims Audience.
+// UID returns the unique ID from the claims ID.
+func (c JWTClaims) UID() (string, error) {
+	if c.RegisteredClaims.ID == "" {
```

Comment:
This will be problematic for Auth0. They do not set jti and explicitly do not allow it from my understanding. They recommend against any jwt id and blacklisting and instead push for short expiration fro access tokens with refresh or silent login in the browser.

I haven't tried it but see: https://community.auth0.com/t/jti-field-doesnt-work/44364

That said the web auth handler can override this and use something else if needed.

Reply:
Resolved. This should be Subject as we talked about.